Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp

```diff
@@ -307,19 +318,45 @@ static gboolean wpeDisplayDRMSetup(WPEDisplayDRM* displayDRM, const char* device
     displayDRM->priv->device = device;
     displayDRM->priv->connector = WTFMove(connector);
 
-    double scale;
-    if (const char* scaleString = getenv("WPE_DRM_SCALE"))
-        scale = g_ascii_strtod(scaleString, nullptr);
-    else {
-        auto* settings = wpe_display_get_settings(WPE_DISPLAY(displayDRM));
-        scale = wpe_settings_get_double(settings, WPE_SETTING_DRM_SCALE, nullptr);
+    static const auto scaleIsInBounds = [](double scale) {
+        return (scale >= 0.05) && (scale <= 20.0);
+    };
+
+    std::optional<double> scaleFromEnvironment;
+    if (const auto scaleString = StringView::fromLatin1(getenv("WPE_DRM_SCALE"))) {
+        RELEASE_ASSERT(scaleString.is8Bit());
+        auto trimmedScaleString = scaleString.trim(isASCIIWhitespace<LChar>);
+        size_t parsedLength = 0;
+        auto scale = parseDouble(trimmedScaleString, parsedLength);
+        if (parsedLength == trimmedScaleString.length() && scaleIsInBounds(scale))
+            scaleFromEnvironment = scale;
```

Comment:
Is all this simpler than `g_ascii_strtod`?  Or is it because it's safer?

Reply:
Easier to follow when not using raw pointers, and harder to make mistakes (which means: a tad safer). Also, if we add to the C-like version the missing whitespace trimming and checking for a successful parse, it turns out to be about the same length:

```cpp
if (const char* scaleString = getenv("WPE_DRM_SCALE")) {
    scaleString = g_strstrip(scaleString);
    char *endptr = NULL;
    auto scale = g_ascii_strtod(scaleString, &endptr);
    if (*endptr == '\0' && (errno != ERANGE) && scaleIsInBounds(scale))
        scaleFromEnvironment = scale;
}
```

In light of this, I think the version proposed in the patch is nicer.